piv/example_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package piv_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"log"
+
+	"github.com/go-piv/piv-go/piv"
+)
+
+func ExampleECDSAPrivateKey_SharedKey_epheremalEncrypt() {
```

Comment:
Couple issues I see with these examples:
* There's no encrypt or decryption here as the name would suggest.
* I don't want to give recommendations on the best ways to use a shared secret for encryption/decryption. There's a lot of ways to get a handshake like this wrong and I'm not a cryptographer.
* The "ExampleECDSAPrivateKey_SharedKey_epheremalEncrypt" doesn't seem like it should belong here, there's nothing piv specific about it.
* Instead of hardwarePublicKey or ephemeralPublicKey, these should probably be publicKey and peerPublicKey.

Can I strongly recommend dropping the examples and doing this as a follow up? It's hard to review PRs that keep adding new code during the review :)

Reply:
Dropped examples.